### PR TITLE
cleanup(generator): separator before code

### PR DESCRIPTION
**Pull request by coryan**
In some places we generated code and then added a separator line in
anticipation of the next code fragment.  That leaves unsightly blank
lines where not needed. In other places we inserted the separator before
the code, which is easier to reason about ("I am about to insert code, I
should add a separator"), and the unneeded lines before a declaration
are not that unsightly, maybe?

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7951)

### CODE REVIEW

**Review by scotthart — APPROVED** [review]

File: generator/integration_tests/golden/mocks/mock_golden_kitchen_sink_connection.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
 class MockGoldenKitchenSinkConnection : public golden::GoldenKitchenSinkConnection {
  public:
+
```
Comment (devbww) [inline generator/integration_tests/golden/mocks/mock_golden_kitchen_sink_connection.h:32]: :-(

Comment (dbolduc) [reply generator/integration_tests/golden/mocks/mock_golden_kitchen_sink_connection.h:32]: +1

Comment (coryan) [reply generator/integration_tests/golden/mocks/mock_golden_kitchen_sink_connection.h]: PTAL.

Comment (dbolduc) [reply generator/integration_tests/golden/mocks/mock_golden_kitchen_sink_connection.h]: :-)

**Review by devbww — APPROVED** [review]

**Pull request merged**